gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/crud_service/access_point/AccessPointCrudServiceImpl.java

```diff
@@ -44,14 +48,23 @@ public void updateAccessPoints(
         final List<AccessPoint> accessPoints
     ) {
         try {
-            accessPointRepository.deleteByReference(AccessPointReferenceType.valueOf(referenceType.name()), referenceId);
-
+            List<io.gravitee.repository.management.model.AccessPoint> deleteAccessPoints = accessPointRepository.deleteByReference(
+                AccessPointReferenceType.valueOf(referenceType.name()),
+                referenceId
+            );
+            if (deleteAccessPoints != null) {
+                deleteAccessPoints
+                    .stream()
+                    .map(AccessPointAdapter.INSTANCE::toEntity)
+                    .forEach(accessPoint -> eventManager.publishEvent(AccessPointEvent.DELETED, accessPoint));
+            }
             for (AccessPoint accessPoint : accessPoints) {
                 var ap = AccessPointAdapter.INSTANCE.fromEntity(accessPoint);
                 if (ap.getId() == null) {
                     ap.setId(UuidString.generateRandom());
                 }
-                accessPointRepository.create(ap);
+                io.gravitee.repository.management.model.AccessPoint createdAccessPoint = accessPointRepository.create(ap);
+                eventManager.publishEvent(AccessPointEvent.CREATED, AccessPointAdapter.INSTANCE.toEntity(createdAccessPoint));
```

Comment:
Should we limit the DELETE events to only the access points we effectively want to delete? Also, considering it is an update, should it be an UPDATE event instead of a CREATE one for those that are already existing?

Reply:
Ok, after discussing, let's keep it simple and delete + recreate the access points.